src/index.js

```diff
@@ -1,25 +1,50 @@
 const puppeteer = require('puppeteer')
 const handlebars = require('handlebars')
 
-module.exports = async function({
-  html,
+module.exports = async function(options) {
+  const {
+    html,
+    content,
+    puppeteerArgs = {},
+  } = options
+
+  if (!html) {
+    throw Error('You must provide an html property.')
+  }
+
+  if (Array.isArray(content)) {
+    const browser = await puppeteer.launch({ ...puppeteerArgs, headless: true })
+
+    const buffers = await Promise.all(content.map(c => {
+      const { output, ...pageContent } = c
+      return makeScreenshot(browser, { ...options, content: pageContent, output })
+    }))
+
+    await browser.close()
+    return buffers
+  }
+
+  const browser = await puppeteer.launch({ ...puppeteerArgs, headless: true })
+  const buffer = await makeScreenshot(browser, options)
+  await browser.close()
+  return buffer
```

Comment:
Instead of `if (Array.isArray())` and duplicating the puppeteer code, I would just build an array there isn't one already:

```suggestion
  const browser = await puppeteer.launch({ ...puppeteerArgs, headless: true })

  const contents = Array.isArray(content) ? content : [content]; // <-- make sure we have an array to map over
  const buffers = await Promise.all(contents.map(c => {
    const { output, ...pageContent } = c
    return makeScreenshot(browser, { ...options, content: pageContent, output })
  }))

  await browser.close()
  return buffers
```

Then you can always map over `contents`, for single images you will just have an array with a single value.

Reply:
I wrote a mixed version. I don't want to introduce a breaking change. In your example having an array with one element when you ask for one image is a breaking change. I think the main usecase is calling node-html-to-image to generate one image. The batch mode is more an optimization. So it needs to remain easy to use for the main usecase. 

This is mixed version doesn't introduce any breaking change and remove the duplication you talked about. What do you think of it?

Reply:
Ah, I think I missed the fact that there is no `content` when a single image is generated 🤦 Sorry about this. It looks great!